java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -765,21 +799,39 @@ private void initAndResizeOffsetBuffer(int currentIndex) {
       }
     }
 
-    private void initValidBuffer() {
+    private void initAndResizeValidBuffer() {
       if (this.valid == null) {
         allocateBitmaskAndSetDefaultValues();
+      } else {
+        long bitmaskSize = ColumnVector.getNativeValidPointerSize((int) rows);
+        if (valid.length < bitmaskSize) {
+          HostMemoryBuffer newValid = HostMemoryBuffer.allocate(bitmaskSize);
+          try {
+            newValid.setMemory(0, bitmaskSize, (byte) 0xFF);
+            newValid.copyFromHostBuffer(0, valid, 0, valid.length);
+            valid.close();
+            valid = newValid;
+            newValid = null;
+          } finally {
+            if (newValid != null) {
+              newValid.close();
+            }
+          }
+        }
       }
     }
 
     private void setNullAt(int index) {
```

Comment:
Could we have some java docs explaining exactly what this does?  Simply because it is not clear that it will call `initAndResizeValidBuffer`, which `appendList` also calls needlessly.

Reply:
Fixed it during refactor